rocAL/rocAL/include/pipeline/master_graph.h

```diff
@@ -305,6 +307,37 @@ template<> inline std::shared_ptr<Cifar10LoaderNode> MasterGraph::add_node(const
     return node;
 }
 
+/*
+ * Explicit specialization for AudioLoaderNode
+ */
+template<> inline std::shared_ptr<AudioLoaderNode> MasterGraph::add_node(const std::vector<rocalTensor*>& inputs, const std::vector<rocalTensor*>& outputs)
+{
+    if(_loader_module)
+        THROW("A loader already exists, cannot have more than one loader")
+    auto node = std::make_shared<AudioLoaderNode>(outputs[0], _device.resources());
+    _loader_module = node->get_loader_module();
+    _loader_module->set_prefetch_queue_depth(_prefetch_queue_depth);
+    _root_nodes.push_back(node);
+    for(auto& output: outputs)
+        _tensor_map.insert(make_pair(output, node));
+
+    return node;
+}
+
+template<> inline std::shared_ptr<AudioLoaderSingleShardNode> MasterGraph::add_node(const std::vector<rocalTensor*>& inputs, const std::vector<rocalTensor*>& outputs)
+{
+    if(_loader_module)
+        THROW("A loader already exists, cannot have more than one loader")
+    auto node = std::make_shared<AudioLoaderSingleShardNode>(outputs[0], _device.resources());
```

Comment:
ENABLE_HIP || ENABLE_OPENCL part is missing check the other similar API

Reply:
Added